ci/ray_ci/doc/cmd_check_api_discrepancy.py

```diff
@@ -1,22 +1,40 @@
 import click
 
 from ci.ray_ci.doc.module import Module
+from ci.ray_ci.doc.autodoc import Autodoc
+from ci.ray_ci.doc.api import AnnotationType
 
 
 @click.command()
 @click.argument("module", required=True, type=str)
-def main(module: str) -> None:
+@click.argument("head_rst_file", required=True, type=str)
+def main(module: str, head_rst_file: str) -> None:
     """
     This script checks for annotated classes and functions in a module, and finds
     discrepancies between the annotations and the documentation.
     """
     module = Module(module)
-    for api in module.get_apis():
-        print(
-            f"API: {api.name}, "
-            f"Annotation Type: {api.annotation_type}, "
-            f"Code Type: {api.code_type}"
-        )
+    api_in_codes = {api.name: api for api in module.get_apis()}
+
+    autodoc = Autodoc(head_rst_file)
+    api_in_docs = {api.name: api for api in autodoc.get_apis()}
+
+    # check that all apis in doc are public api
+    print("Finding invalid APIs in doc:\n")
+    for api_name, api_in_doc in api_in_docs.items():
+        api_in_code = api_in_codes.get(api_name)
+        if not api_in_code or api_in_code.annotation_type != AnnotationType.PUBLIC_API:
+            print(f"\t{api_in_doc}")
+    print("\n")
+
+    # check that all public apis are in doc
+    print("Finding missing APIs in doc:\n")
+    for api_name, api_in_code in api_in_codes.items():
+        if (
+            api_in_code.annotation_type == AnnotationType.PUBLIC_API
+            and api_name not in api_in_docs
+        ):
+            print(f"\t{api_in_code}")
```

Comment:
do we want to return 2 list of APIs instead of printing it out?

Reply:
^ might be easier to parse for other work but also if you just want to take a look at which one doesn't meet the bar I think that's also fine